src/LinAlg/hiopMatrix.hpp

```diff
@@ -84,6 +84,16 @@ class hiopMatrix
   virtual hiopMatrix* alloc_clone() const=0;
   virtual hiopMatrix* new_copy() const=0;
 
+  virtual void copyFrom(const hiopMatrix& matrix) = 0;
+  virtual void copyFrom(const double* buffer) = 0;
```

Comment:
this method does not make sense for sparse of mds matrices (specific to dense matrices in other words). I would not make it part of the interface.